### ISSUE TITLE
provide serializer API function to undo value preamble

### ISSUE THREAD

**Issue by WebFreak001** [libmir/mir-ion#13]
I have a custom type like this:
```d
struct NoneType {}
alias Optional(T) = MySumType!(NoneType, T);

// calling `match` on this custom type is blocked by https://issues.dlang.org/show_bug.cgi?id=21975
struct MySumType(Types...)
{
	SumType!Types value;
	alias value this;

	ref typeof(this) opAssign(T)(T rhs)
	{
		value.opAssign(rhs);
		return this;
	}

	void serialize(S)(ref S serializer) const
	{
		import mir.ser : serializeValue;

		value.tryMatch!(
			(NoneType none) { /* no output */ },
			(v) { serializeValue(serializer, v); }
		);
	}
}

struct S { Optional!bool optBool; }
```

and want it to serialize to completely omit these values, however currently it serializes like this:

```
{"optBool":}
```

I would like to be able to check if a key was written and undo the key write in that case. In my use-case it's not possible to have such a value in an array or as standalone serialization, but having some kind of generic way to undo preparations (leading comma, etc) would be nice anyway.

I would work around this by only selectively implementing the JsonSerializer and accessing its internal appender, however it's private so that's not possible. A function to undo the last added key (and reverting state so that commas are properly placed) would be nice to have.

This is a very frequent case in my code base / the protocol I implement, where omission of fields often means something different than set default values or maybe don't even have default values that are predictable before actually having context.
